src/s1reader/s1_orbit.py

```diff
@@ -1,12 +1,171 @@
+    pad_end_time = end_time + pad_30_min
+    new_start_time = pad_start_time.strftime('%Y-%m-%dT%H:%M:%S')
+    new_end_time = pad_end_time.strftime('%Y-%m-%dT%H:%M:%S')
+    query_string = f"startswith(Name,'S1{sensor_id}') and substringof('{orbit_type}',Name) " \
+                   f"and ContentDate/Start lt datetime'{new_start_time}' and ContentDate/End gt datetime'{new_end_time}'"
+    query_params = {'$top': 1, '$orderby': 'ContentDate/Start asc',
+                    '$filter': query_string}
+    query_response = requests.get(url=scihub_url, params=query_params,
+                                  auth=(scihub_user, scihub_password))
+    # Parse XML tree from query response
+    xml_tree = ElementTree.fromstring(query_response.content)
+    # Extract w3.org URL
+    w3_url = xml_tree.tag.split('feed')[0]
+
+    # Extract orbit's name, id, url
+    orbit_id = xml_tree.findtext(
+        f'.//{w3_url}entry/{m_url}properties/{d_url}Id')
+    orbit_url = f"{scihub_url}('{orbit_id}')/$value"
+    orbit_name = xml_tree.findtext(f'./{w3_url}entry/{w3_url}title')
+
+    if orbit_id is not None:
+        orbit_dict = {'orbit_name': orbit_name, 'orbit_type': orbit_type,
+                      'orbit_url': orbit_url}
+    else:
+        orbit_dict = None
+    return orbit_dict
+
+
+def download_orbit_file(output_folder, orbit_url):
+    '''
+    Download S1-A/B orbits
+    Parameters:
+    ----------
+    output_folder: str
+        Path to directory where to store orbits
+    orbit_url: str
+        Remote url of orbit file to download
+    '''
+
+    response = requests.get(url=orbit_url, auth=(scihub_user, scihub_password))
+    # Get header and find filename
+    header = response.headers['content-disposition']
+    _, header_params = cgi.parse_header(header)
+    # construct orbit filename
+    orbit_filename = os.path.join(output_folder, header_params['filename'])
+    # Save orbits
+    open(orbit_filename, 'wb').write(response.content)
+
+
+def download_orbit(safe_file, orbit_dir):
+    '''
+    Download orbits for S1-A/B SAFE "safe_file"
+
+    Parameters
+    ----------
+    safe_file: str
+        File path to SAFE file for which download the orbits
+    orbit_dir: str
+        File path to directory where to store downloaded orbits
+    '''
+
+    # Create output directory & check internet connection
+    os.makedirs(orbit_dir, exist_ok=True)
+    check_internet_connection()
+
+    # Parse info from SAFE file name
+    sensor_id, _, start_time, end_time, _ = parse_safe_filename(safe_file)
+
+    # Find precise orbit first
+    orbit_dict = get_orbit_dict(sensor_id, start_time,
+                                end_time, 'AUX_POEORB')
+    # If orbit dict is empty, find restituted orbits
+    if orbit_dict == None:
+        orbit_dict = get_orbit_dict(sensor_id, start_time,
+                                    end_time, 'AUX_RESORB')
+    # Download orbit file
+    download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
```

Comment:
Do we want to add a check here that the file already exists so we don't redownload things unnecessarily? Could be 
```python
    filename = os.path.join(orbit_dir, orbit_dict["orbit_name"])
    if not os.path.exists(filename):
        download_orbit_file(orbit_dir, orbit_dict["orbit_url"])
```